.github/workflows/ci.yml

```diff
@@ -83,7 +83,7 @@ jobs:
             cflags: '-DR2_ASSERT_STDOUT=1'
             meson_options: -Db_sanitize=address,undefined
             asan: true
-            asan_options: 'detect_leaks=0,detect_odr_violation=0'
+            asan_options: 'detect_leaks=0,detect_odr_violation=0,allocator_may_return_null=1'
```

Comment:
Wasn't the original plan to clamp the allocation so that any useful information in the binary can be salvaged even though the binary is malformed? Not that I disapprove of this but others might.

Reply:
yes. malformed binaries should be loaded

Reply:
but i dont understand whats this asan option about. it is assuming that malloc can return null?

Reply:
When you enable ASAN, if you do a malloc with a size larger than X (i think X by default is 0x100000000 or something like that) ASAN will complain about the size being too large. Now, what happens in real world, without ASAN, is that when you pass such a large value malloc will just return NULL. This option prevent ASAN from blocking the program execution when such malloc-"errors" happen.

Reply:
so with this enabled, malloc will return null instead of warning and continue the execution?

Reply:
yes